Sources/TuistGenerator/Settings/DefaultSettingsProvider.swift

```diff
@@ -115,3 +116,36 @@ public final class DefaultSettingsProvider: DefaultSettingsProviding {
         }
     }
 }
+
+enum BuildSettingsError: FatalError {
+    case invalidValue(Any)
+
+    var description: String {
+        switch self {
+        case let .invalidValue(value):
+            return "Cannot convert \"\(value)\" to SettingValue type"
```

Comment:
As a user being presented with this error, I wouldn't know what to do? I'd provide them with something more actionable:

```
The build setting 'mySetting' has an invalid value "xx" of type string, which is not supported. The supported types are: string, integer, array...
```